frontend/src/metabase/sharing/components/EmailAttachmentPicker.jsx

```diff
@@ -20,140 +23,163 @@ export default class EmailAttachmentPicker extends Component {
     setPulse: PropTypes.func.isRequired,
   };
 
-  DEFAULT_ATTACHMENT_TYPE = "csv";
+  componentDidMount() {
+    this.setState(this.calculateStateFromCards());
+  }
 
-  updateStateFromCards() {
-    const { cards } = this.props;
-    const { attachmentType } = this.state;
+  componentDidUpdate() {
+    const newState = this.calculateStateFromCards();
+    
+    newState.isEnabled = newState.isEnabled || this.state.isEnabled;
+    if(newState.selectedCardIds.size == 0) newState.selectedAttachmentType = this.state.selectedAttachmentType;
```

Comment:
these two lines especially feel like a hack, but I didn't know what to do about it.

The issue is that if no cards have been selected and the user clicks the ".xls" button, the calculated state is an attachment type of ".csv" (the default—there are no selected cards with `include_xls: true` to say otherwise).